feat(annotations): Allow a comment at the end of a tflint-ignore annotation (#1892)
terraform-linters/tflint: docs/user-guide/annotations.md

    instance_type = "t1.2xlarge"
}
```

It is possible to add a reason for why a rule is ignored:

```hcl
resource "aws_instance" "foo" {
    # tflint-ignore: aws_instance_invalid_type # A good reason
    instance_type = "t1.2xlarge"
}

> ```

Reviewer: It would be nice to mention what types of comment formats are allowed.
In my understanding, you can add comments after the `#` and `//` at the end of the annotation comment.

Strictly speaking, comments starting with `*` or `/` are probably valid for reasons of regexp, but I would like to make it clear that this is not what we intended.

> ```

Reviewer 2: When placing the annotation on the previous line, also breaking the comment onto its own line(s) seems preferable. But this is a nice addition for users of same-line annotations, or very short comments.

GitHub PR suggestions don't allow for back-ticked blocks inside, here's what I have in mind:

<hr>

It is possible to add a reason for why a rule is ignored:

```hcl
resource "aws_instance" "foo" {
    # This instance type is new and TFLint doesn't know about it yet
    # tflint-ignore: aws_instance_invalid_type
    instance_type = "t10.2xlarge"
}
```

Or, on the same line:

```hcl
resource "aws_instance" "foo" {
    instance_type = "t10.2xlarge" # tflint-ignore: aws_instance_invalid_type # too new for TFLint
}
```

Author: > GitHub PR suggestions don't allow for back-ticked blocks inside, here's what I have in mind:

It actually does work! (Thanks @FluffyBucket for showing me!) You just need to use more than 3 backticks on the outermost block.

````suggestion
It is possible to add a reason for why a rule is ignored:

```terraform
resource "aws_instance" "foo" {
    # This instance type is new and TFLint doesn't know about it yet
    # tflint-ignore: aws_instance_invalid_type
    instance_type = "t10.2xlarge"
}
```

Or, on the same line:

```terraform
resource "aws_instance" "foo" {
    instance_type = "t10.2xlarge" # tflint-ignore: aws_instance_invalid_type # too new for TFLint
}
```
````

I'll add this, together with @wata727's suggestion about `//`

    instance_type = "t1.2xlarge"
}
```

> It is possible to add a reason for why a rule is ignored:

Reviewer 2: ```suggestion
It's a good idea to add a reason for why a rule is ignored, especially temporarily:
```

```hcl
resource "aws_instance" "foo" {
    # This instance type is new and TFLint doesn't know about it yet
    # tflint-ignore: aws_instance_invalid_type
    instance_type = "t10.2xlarge"
}
```

Or, on the same line:

```hcl
resource "aws_instance" "foo" {
  # tflint-ignore: aws_instance_invalid_type # too new for TFLint
  instance_type = "t10.2xlarge" 
}
```

> One can also use `//`-style comments

Reviewer 2: ```suggestion
The `//` comment style is also supported, but Terraform recommends `#`.
```
